streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -358,22 +420,40 @@ public void writeAsText(final String filePath,
         for (final Predicate<? super K, ? super V> predicate : predicates) {
             Objects.requireNonNull(predicate, "predicates can't have null values");
         }
+        final String branchName = builder.newProcessorName(BRANCH_NAME);
+        final String[] childNames = new String[predicates.length];
+        final List<StreamsGraphNode> branchChildNodes = new ArrayList<>();
 
-        String branchName = builder.newProcessorName(BRANCH_NAME);
-
-        String[] childNames = new String[predicates.length];
+        KStream<K, V>[] branchChildren = (KStream<K, V>[]) Array.newInstance(KStream.class, predicates.length);
```

Comment:
nit: can be final.

Reply:
General question: Do we need to use arrays in these builders?

Using the java collections libraries would give us better generics support so we wouldn't need to cast as much. I think these objects are all going to be eligible for GC once the topology is actually built, therefore they aren't going to live during run time, so heap shouldn't be an issue.